bundle/phases/bind.go

```diff
@@ -1,45 +1,57 @@
 package phases
 
 import (
+	"context"
+
 	"github.com/databricks/cli/bundle"
 	"github.com/databricks/cli/bundle/deploy/lock"
 	"github.com/databricks/cli/bundle/deploy/terraform"
+	"github.com/databricks/cli/libs/diag"
+	"github.com/databricks/cli/libs/log"
 )
 
-func Bind(opts *terraform.BindOptions) bundle.Mutator {
-	return newPhase(
-		"bind",
-		[]bundle.Mutator{
-			lock.Acquire(),
-			bundle.Defer(
-				bundle.Seq(
-					terraform.StatePull(),
-					terraform.Interpolate(),
-					terraform.Write(),
-					terraform.Import(opts),
-					terraform.StatePush(),
-				),
-				lock.Release(lock.GoalBind),
-			),
-		},
-	)
+func Bind(ctx context.Context, b *bundle.Bundle, opts *terraform.BindOptions) diag.Diagnostics {
+	log.Info(ctx, "Phase: bind")
+
+	diags := bundle.Apply(ctx, b, lock.Acquire())
+	if diags.HasError() {
+		return diags
+	}
+
+	defer func() {
+		diags = diags.Extend(bundle.Apply(ctx, b, lock.Release(lock.GoalBind)))
```

Comment:
Defer called after return, right? So extending diags won't have an effect on the slice?

Reply:
No, defer is called before return, so it should be included: https://go.dev/play/p/UL60FF1WehT

Reply:
It seems like if `var diags []string` is defined inside teh function, it does not
https://go.dev/play/p/gPd6A1wIq-h

Reply:
Good catch, fixed - b41af7caba261192e537e82c3b0975d57ebcf106